tests/ConnectionTest.php

```diff
@@ -134,6 +137,114 @@ public function testConnectDispatchEvent(): void
         $conn->connect();
     }
 
+    public function testTransactionBeginDispatchEvent(): void
+    {
+        $listenerMock = $this->createMock(TransactionBeginDispatchEventListener::class);
+        $listenerMock->expects(self::exactly(2))->method('onTransactionBegin');
+
+        $eventManager = new EventManager();
+        $eventManager->addEventListener([Events::onTransactionBegin], $listenerMock);
+        $listener = new class {
```

Comment:
Since this is a connection unit test, I'd prefer to have the listener interface mocked instead of being implemented with additional code. See other tests for example:

https://github.com/doctrine/dbal/blob/c7201157f8e567fe4b61b0ffa0f1f04c0dc19992/tests/ConnectionTest.php#L124-L127

Reply:
How can I `self::assertSame($conn, $listener->getConnection());` with mock?

Reply:
I saw it... it have 
```php
$this->callback(function($subject){
  return is_callable([$subject, 'getName']) &&
         $subject->getName() == 'My subject';
})
```